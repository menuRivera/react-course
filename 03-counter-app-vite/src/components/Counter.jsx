import PropTypes from 'prop-types'
import { useState } from 'react';

const handleEvent = (e) => {
    console.log(e);
}

export const Counter = ({ value }) => {
    const [counter, setCounter] = useState(value)

    const handleAdd = () => setCounter(counter + 1)
    const handleSubstract = () => setCounter(counter - 1)
    const handleReset = () => setCounter(value)

    return (
        <>
            <h1>Counter</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}>+1</button>
            <button aria-label='btn-reset' onClick={handleSubstract}>-1</button>
            <button onClick={handleReset}>Reset</button>
        </>
    )
}

Counter.propTypes = {
    value: PropTypes.number.isRequired
}