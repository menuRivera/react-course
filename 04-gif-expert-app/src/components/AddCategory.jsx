import React from 'react'
import { useState } from 'react'

export const AddCategory = ({ onNewValue }) => {
    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()
        if (inputValue.trim().length <= 1) return

        // setCategories(cat => [...cat, inputValue])
        onNewValue(inputValue.trim())
        setInputValue('')
    }
    return (
        <form onSubmit={onSubmit}>
            <input type="text" placeholder='Buscar gifs' value={inputValue} onChange={onInputChange} />
        </form>
    )
}
