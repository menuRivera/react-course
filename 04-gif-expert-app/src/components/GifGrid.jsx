import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs'
import { GifItem } from './GifItem'

export const GifGrid = ({ category }) => {
    const [images, setImages] = useState([])

    const getImages = async () => {
        const newImgs = await getGifs(category)
        setImages(newImgs)
    }

    useEffect(() => {
        // Esto esta bien, pero no es tan común
        // getGifs(category)
        //     .then(imgs => {
        //         setImages(imgs)
        //     })
        getImages() // Esto es más común
    }, [])

    return (
        <>
            <h3>{category}</h3>
            {/* {images} */}
            <div className='card-grid'>
                {
                    images.map(img => (
                        // <li key={img.id}>
                        //     {img.title}
                        // </li>
                        <GifItem key={img.id} {...img} />
                    ))
                }
            </div>
        </>
    )
}
