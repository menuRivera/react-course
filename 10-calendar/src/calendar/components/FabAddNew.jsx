import { addHours } from "date-fns/esm"
import { useCalendarStore } from "../../hooks/useCalendarStore"
import { useUiStore } from "../../hooks/useUiStore"

export const FabAddNew = () => {

    const { openDateModal } = useUiStore()
    const { setActiveEvent } = useCalendarStore()

    const handleClickNew = () => {
        setActiveEvent({
            title: '',
            notes: '',
            start: new Date(),
            end: addHours(new Date(), 2),
            bgColor: '#fafafa',
            user: {
                _id: 'fdsafa',
                name: 'Manuel'
            }
        })
        openDateModal()
    }

    return (
        <button className="btn btn-primary fab" onClick={handleClickNew}>
            <i className="fas fa-plus"></i>
        </button>
    )
}