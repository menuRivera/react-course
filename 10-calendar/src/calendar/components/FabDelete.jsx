import { useCalendarStore } from "../../hooks/useCalendarStore"

export const FabDelete = () => {
    const { startDeletingEvent, hasEventSelected } = useCalendarStore()

    const handleDelete = () => {
        startDeletingEvent()
    }

    if (!hasEventSelected) return <></>

    return (
        <button
            aria-label="btn-delete"
            className="btn btn-danger fab-danger"
            onClick={handleDelete}>
            <i className="fas fa-trash-alt"></i>
        </button>
    )
}