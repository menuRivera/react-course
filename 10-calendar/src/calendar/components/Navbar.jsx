import { useAuthStore } from "../../hooks/useAuthStore"

export const Navbar = () => {

    const { startLogout, user } = useAuthStore()

    return (
        <div className="navbar navbar-dark bg-dark text-white mb-4 px-4">
            <span>
                <i className="fas fa-calendar-alt">
                    &nbsp;
                    {user.name}
                </i>
            </span>

            <button className="btn btn-outline-danger" onClick={startLogout}>
                <i className="fas fa-sign-out-alt"></i>
                &nbsp;
                <span>Salir</span>
            </button>
        </div>
    )
}