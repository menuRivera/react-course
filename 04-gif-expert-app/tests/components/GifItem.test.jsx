import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem/>', () => {
    const title = 'Goku';
    const url = 'https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=5d6963d424eo07zs9ycmcc7ndeh51rmaltkmm5wsdnjyt9ht&rid=giphy.gif&ct=g'


    test('Debe evaluar el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)

        expect(container).toMatchSnapshot()
    })

    test('Debe mostrar la imÃ¡gen con el url y el alt indicado', () => {
        render(<GifItem title={title} url={url} />)

        const { src, alt } = screen.getByRole('img')

        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />)

        expect(screen.getByText(title)).toBeTruthy()
    })
})