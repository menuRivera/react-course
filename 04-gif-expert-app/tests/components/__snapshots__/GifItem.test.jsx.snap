// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe evaluar el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=5d6963d424eo07zs9ycmcc7ndeh51rmaltkmm5wsdnjyt9ht&rid=giphy.gif&ct=g"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
